### DISCUSSION TITLE
discussions: GitHub emojis render as `undefined`

### DISCUSSION THREAD

**Discussion by cherryblossom000** [lowlighter/metrics#740] [Q&A] [answered]
[Example](https://gist.githubusercontent.com/cherryblossom000/f34ce5b1bc70befdb2502e3292e2f191/raw/96552cb4862cdf042884ec5e69c78a25051ea01f/github-metrics.svg)

![discussions section of rendered SVG, showing ‘undefined General Feedback’](https://user-images.githubusercontent.com/31467609/148322840-03b645bc-ffd0-4a25-91b4-e02afc2b5e14.png)

The emoji of the [General Feedback category](https://github.com/github/feedback/discussions/categories/general-feedback) is `:octocat:`. This is rendered as `undefined`, even if `config_gemoji: true`.

Possible fix:

https://github.com/lowlighter/metrics/blob/63a84e24746be9e44770611cd298c060d24526a3/source/plugins/discussions/index.mjs#L36

```diff
- for (const category of [...fetched.map(({category:{emoji, name}}) => `${imports.emoji.get(emoji)} ${name}`)]) 
+ for (const category of fetched.map(({category:{emoji, name}}) => `${imports.emoji.get(emoji) ?? emoji} ${name}`))
```

**Comment by lowlighter** [comment, marked as answer]
Thanks for your report 👍 !
Will be tracked in #747
